africanus/dft/examples/predict_from_fits.py

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+
+import argparse
+
+import numpy as np
+from astropy.io import fits
+import dask
+import dask.array as da
+from dask.diagnostics import ProgressBar
+from africanus.dft.dask import im_to_vis
+import xarray as xr
+from xarrayms import xds_from_ms, xds_from_table, xds_to_table
+
+
+def create_parser():
+    p = argparse.ArgumentParser()
+    p.add_argument("--ms", type=str)
+    p.add_argument("--fitsmodel", type=str)
+    p.add_argument("--row_chunks", default=10000, type=int)
+    p.add_argument("--ncpu", default=0, type=int)
+    p.add_argument("--colname", default="MODEL_DATA", type=str)
+    return p
+
+
+args = create_parser().parse_args()
+
+if args.ncpu:
+    ncpu = args.ncpu
+    from multiprocessing.pool import ThreadPool
+    dask.config.set(pool=ThreadPool(ncpu))
+else:
+    import multiprocessing
+    ncpu = multiprocessing.cpu_count()
+
+print("Using %i threads" % ncpu)
+
+# Get MS frequencies
+spw_ds = list(xds_from_table("::".join((args.ms, "SPECTRAL_WINDOW")),
+                             group_cols="__row__"))[0]
+
+# Get frequencies in the measurement set
+# If these do not match those in the fits
+# file we need to interpolate
+ms_freqs = spw_ds.CHAN_FREQ.data
+nchan = ms_freqs.compute().size
+
+# load in the fits file
+model = fits.getdata(args.fitsmodel)
+# get header
+hdr = fits.getheader(args.fitsmodel)
+
+# get image coordinates
+npix_l = hdr['NAXIS1']
+refpix_l = hdr['CRPIX1']
+delta_l = hdr['CDELT1'] * np.pi/180  # assumes untis are deg
+l_coord = np.arange(1 - refpix_l, 1 + npix_l - refpix_l)*delta_l
+
+npix_m = hdr['NAXIS2']
+refpix_m = hdr['CRPIX2']
+delta_m = hdr['CDELT2'] * np.pi/180  # assumes untis are deg
+m_coord = np.arange(1 - refpix_m, 1 + npix_m - refpix_m)*delta_m
+
+npix_tot = npix_l * npix_m
+
+# get frequencies
+nband = hdr['NAXIS4']
+refpix_nu = hdr['CRPIX3']
```

Comment:
Should this not be CRPIX4?

What's on AXIS3, correlation?

Reply:
Yep, well spotted. They just happen to be the same for my test case. Thanks